File: sdk/javascript/src/execution.ts
import { PlatformClient } from "./client";

export interface Execution {
  id: string;
  projectId: string;
  status: string;
  startedAt: string;
  finishedAt?: string;
}

export interface ExecutionLogs {
  executionId: string;
  logs: string[];
}

export class ExecutionAPI {
  constructor(private readonly client: PlatformClient) {}

  list() {
    return this.client.request<Execution[]>({
      method: "GET",
      url: "/executions"
    });
  }

  get(executionId: string) {
    return this.client.request<Execution>({
      method: "GET",
      url: `/executions/${executionId}`
    });
  }

  cancel(executionId: string) {
    return this.client.request<void>({
      method: "POST",
      url: `/executions/${executionId}/cancel`
    });
  }

  logs(executionId: string) {
    return this.client.request<ExecutionLogs>({
      method: "GET",
      url: `/executions/${executionId}/logs`
    });
  }
}

File: sdk/python/ai_platform_sdk/execution.py
class ExecutionAPI:
    def __init__(self, client):
        self.client = client

    def create(self, payload):
        return self.client.post(
            "/executions",
            payload,
        )

    def get(self, execution_id):
        return self.client.get(
            f"/executions/{execution_id}"
        )

    def cancel(self, execution_id):
        return self.client.post(
            f"/executions/{execution_id}/cancel"
        )